### PR TITLE
[Fix](Short Circuit) fix no project list in OlapScanNode

### PR DESCRIPTION
**Pull request by eldenmoon**
query like

```
select a, `__DORIS_DELETE_SIGN__` from  `test_tbl` WHERE k = '1111111'
```
will not contain project list on top of OlapScanNode, so for short circuit queries, we should use output exprs on top of plan fragment.

```
+----------------------------------------------------------------------------------------------------------------------------+
| Explain String(Nereids Planner)                                                                                            |
+----------------------------------------------------------------------------------------------------------------------------+
| PLAN FRAGMENT 0                                                                                                            |
|   OUTPUT EXPRS:                                                                                                            |
|     k1[#0]                                                                                                                 |
|     k2[#1]                                                                                                                 |
|     k3[#2]                                                                                                                 |
|     k4[#3]                                                                                                                 |
|     k5[#4]                                                                                                                 |
|     k6[#5]                                                                                                                 |
|     k7[#6]                                                                                                                 |
|     k8[#7]                                                                                                                 |
|     k9[#8]                                                                                                                 |
|     k10[#9]                                                                                                                |
|     k11[#10]                                                                                                               |
|     k12[#11]                                                                                                               |
|     new_column0[#12]                                                                                                       |
|     new_column1[#13]                                                                                                       |
|     __DORIS_DELETE_SIGN__[#14]                                                                                             |
|   PARTITION: UNPARTITIONED                                                                                                 |
|                                                                                                                            |
|   HAS_COLO_PLAN_NODE: false                                                                                                |
|                                                                                                                            |
|   VRESULT SINK                                                                                                             |
|      MYSQL_PROTOCAL                                                                                                        |
|                                                                                                                            |
|   1:VEXCHANGE                                                                                                              |
|      offset: 0                                                                                                             |
|      distribute expr lists: k1[#0], k2[#1], k3[#2]                                                                         |
|                                                                                                                            |
| PLAN FRAGMENT 1                                                                                                            |
|                                                                                                                            |
|   PARTITION: HASH_PARTITIONED: k1[#0], k2[#1], k3[#2]                                                                      |
|                                                                                                                            |
|   HAS_COLO_PLAN_NODE: false                                                                                                |
|                                                                                                                            |
|   STREAM DATA SINK                                                                                                         |
|     EXCHANGE ID: 01                                                                                                        |
|     UNPARTITIONED                                                                                                          |
|                                                                                                                            |
|   0:VOlapScanNode(149)                                                                                                     |
|      TABLE: regression_test_serving_p0.tbl_point_query0(tbl_point_query0), PREAGGREGATION: ON                              |
|      PREDICATES: (k1[#0] = 1231) AND (k2[#1] = 119291.110000000) AND (k3[#2] = 'ddd') AND (__DORIS_DELETE_SIGN__[#14] = 0) |
|      partitions=1/1 (tbl_point_query0)                                                                                     |
|      tablets=1/1, tabletList=13203                                                                                         |
|      cardinality=0, avgRowSize=0.0, numNodes=1                                                                             |
|      pushAggOp=NONE                                                                                                        |
|      SHORT-CIRCUIT                                                                                                         |
|                                                                                                                            |
|                                                                                                                            |
| Statistics                                                                                                                 |
|  planed with unknown column statistics                                                                                     |
+----------------------------------------------------------------------------------------------------------------------------+
```

## Proposed changes

Issue Number: close #xxx

### CODE REVIEW

**Comment by eldenmoon** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/qe/ShortCircuitQueryContext.java [hunk 1]
```diff
@@ -66,8 +67,15 @@ public ShortCircuitQueryContext(Planner planner, Queriable analzyedQuery) throws
         this.serializedQueryOptions = ByteString.copyFrom(
                 new TSerializer().serialize(options));
         List<TExpr> exprs = new ArrayList<>();
-        for (Expr expr : planner.getFragments().get(1).getPlanRoot().getProjectList()) {
-            exprs.add(expr.treeToThrift());
+        OlapScanNode olapScanNode = (OlapScanNode) planner.getFragments().get(1).getPlanRoot();
+        if (olapScanNode.getProjectList() != null) {
+            // project on scan node
+            exprs.addAll(olapScanNode.getProjectList().stream()
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/qe/ShortCircuitQueryContext.java:73]: should think about
```
result sink([#3, #2, #1])
+-- project([#1, #2, #3])
    +-- scan
```

Comment (eldenmoon) [reply fe/fe-core/src/main/java/org/apache/doris/qe/ShortCircuitQueryContext.java:73]: ok, i will refactor some of the code after legacy planner removed

**Review by morrySnow — APPROVED** [review]

**Review by lxr599 — APPROVED** [review]
LGTM

**Pull request merged**